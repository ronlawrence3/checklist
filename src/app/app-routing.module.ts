import { NgModule } from '@angular/core';
import { Routes, RouterModule } from '@angular/router';
import { HomeComponent } from './home/home.component';

const routes: Routes = [
  {
    path: 'lists',
    loadChildren: () => import('./checklist/checklist.module').then(mod => mod.ChecklistModule),
  },
  {
    path: 'home',
    component: HomeComponent,
  },
  { path: '', redirectTo: 'home', pathMatch: 'full' },
  { path: '**', component: HomeComponent },
];

@NgModule({
  imports: [RouterModule.forRoot(routes)],
  exports: [RouterModule],
})
export class AppRoutingModule {}
