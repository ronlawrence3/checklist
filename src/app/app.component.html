<mat-toolbar color="primary" class="navbar-formatting">
  <span class="navbar-title" routerLink="/">Checklists!</span>
  <button mat-flat-button color="primary" routerLink="/lists" routerLinkActive="nav-active">Lists</button>
  <button mat-flat-button color="primary" routerLink="/admin" routerLinkActive="nav-active">Admin</button>
  <span class="gap"></span>
  <button mat-flat-button color="primary" routerLink="/help" routerLinkActive="nav-active">Help</button>
</mat-toolbar>
<div class="content-container">
  <router-outlet></router-outlet>
</div>
