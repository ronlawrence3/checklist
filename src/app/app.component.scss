.navbar-title {
  cursor: pointer;
  padding-right: 20px;
}
.navbar-formatting {
  display: flex;
  justify-content: space-evenly;

}

.nav-active {
  background-blend-mode: color-burn;
  border: 1px solid;
}

.gap {
  flex-grow: 1;
}
