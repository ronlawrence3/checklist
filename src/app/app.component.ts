import { Component, OnDestroy, OnInit } from '@angular/core';
import { select, Store } from '@ngrx/store';
import { AppState } from './reducers';
import { Router } from '@angular/router';
import { selectAllChecklists } from './store/checklists/check-list/check-list.selectors';
import { Subscription } from 'rxjs';
import { loadCheckLists } from './store/checklists/check-list/check-list.actions';
import { CheckList } from './store/checklists/check-list/check-list.model';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss'],
})
export class AppComponent implements OnInit, OnDestroy {
  title = 'checklist';
  sideNavOpened = false;
  allLists: CheckList[] = [];
  allLists$ = this.store.pipe(select(selectAllChecklists));
  private subscription = new Subscription();

  constructor(private store: Store<AppState>, private router: Router) {}

  ngOnInit() {
    this.subscription.add(
      this.allLists$.subscribe(lists => {
        this.allLists = lists as any;
      })
    );
    const list1: CheckList = {
      id: 'list1',
      name: 'List 1',
      items: [
        { checked: true, id: '1', text: 'First' },
        { checked: false, id: '2', text: 'Two' },
        { checked: false, id: '3', text: 'three' },
        { checked: false, id: '4', text: 'Four' },
        { checked: false, id: '5', text: 'Five' },
        { checked: false, id: '6', text: 'six' },
        { checked: false, id: '7', text: 'seven' },
        { checked: false, id: '8', text: 'eight' },
        { checked: false, id: '9', text: 'Nine' },
        { checked: false, id: '10', text: '' },
        { checked: false, id: '11', text: '' },
      ],
      completedItems: [],
    };
    this.store.dispatch(loadCheckLists({ checkLists: [list1] }));
  }

  ngOnDestroy() {
    this.subscription.unsubscribe();
  }

  addList() {
    console.log('here we will add a new list');
  }

  openList(list: CheckList) {
    this.router.navigate([`/lists/${list.id}`]);
  }
}
