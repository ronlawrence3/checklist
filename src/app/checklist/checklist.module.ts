import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ChecklistListComponent } from './checklist-list/checklist-list.component';
import { ChecklistComponent } from './checklist/checklist.component';
import { ChecklistItemComponent } from './checklist-item/checklist-item.component';
import { RouterModule, Routes } from '@angular/router';
import { MatCheckboxModule } from '@angular/material/checkbox';
import { MatInputModule } from '@angular/material/input';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { MatCardModule } from '@angular/material/card';
import { MatButtonModule } from '@angular/material/button';
import { MatIconModule } from '@angular/material/icon';
import { DisplayChecklistComponent } from './display-checklist/display-checklist.component';

const routes: Routes = [
  {
    path: '',
    component: ChecklistListComponent,
  },
  {
    path: ':id',
    component: DisplayChecklistComponent,
  },
];

@NgModule({
  declarations: [ChecklistListComponent, ChecklistComponent, ChecklistItemComponent, DisplayChecklistComponent],
  exports: [ChecklistItemComponent, ChecklistListComponent],
  imports: [
    CommonModule,
    RouterModule.forChild(routes),
    MatCheckboxModule,
    MatInputModule,
    FormsModule,
    ReactiveFormsModule,
    MatCardModule,
    MatButtonModule,
    MatIconModule,
  ],
})
export class ChecklistModule {}
