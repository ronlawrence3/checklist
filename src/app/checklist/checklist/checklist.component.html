<div [formGroup]="checklistForm">
  <mat-form-field color="primary">
    <input matInput formControlName="name" />
  </mat-form-field>
  <mat-expansion-panel togglePosition="before">
    <mat-expansion-panel-header>Description ...</mat-expansion-panel-header>
    <mat-form-field>
      <textarea matInput formControlName="description"></textarea>
    </mat-form-field>
  </mat-expansion-panel>

  <div class="list-container">
    <div *ngFor="let i of checkList?.items" class="item-container">
      <app-checklist-item [item]="i" (itemChanged)="itemChanged($event, i)"></app-checklist-item>
    </div>
  </div>
  <mat-expansion-panel>
    <mat-expansion-panel-header>Completed:</mat-expansion-panel-header>
    <div *ngFor="let i of checkList?.completedItems">
      <app-checklist-item [item]="i" (itemChanged)="completedItemChanged($event, i)"></app-checklist-item>
    </div>
  </mat-expansion-panel>
</div>
