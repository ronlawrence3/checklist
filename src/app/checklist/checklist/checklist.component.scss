.list-container {
  display: flex;
  flex-direction: column;
  width: 100%;
}

.item-container {
  display: flex;
  flex-direction: row;
  width: 100%;
}
