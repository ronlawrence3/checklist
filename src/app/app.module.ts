import { BrowserModule } from '@angular/platform-browser';
import { NgModule } from '@angular/core';

import { AppRoutingModule } from './app-routing.module';
import { AppComponent } from './app.component';
import { StoreModule } from '@ngrx/store';
import { reducers, metaReducers } from './reducers';
import { EffectsModule } from '@ngrx/effects';
import { AppEffects } from './app.effects';
import { StoreRouterConnectingModule } from '@ngrx/router-store';
import { EntityDataModule } from '@ngrx/data';
import { entityConfig } from './entity-metadata';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { ServiceWorkerModule } from '@angular/service-worker';
import { environment } from '../environments/environment';

@NgModule({
  declarations: [
    AppComponent
  ],
  imports: [
    BrowserModule,
    AppRoutingModule,
    StoreModule.forRoot(reducers, {
      metaReducers,
      runtimeChecks: {
        strictStateImmutability: true,
        strictActionImmutability: true
      }
    }),
    EffectsModule.forRoot([AppEffects]),
    StoreRouterConnectingModule.forRoot(),
    EntityDataModule.forRoot(entityConfig),
    BrowserAnimationsModule,
    ServiceWorkerModule.register('ngsw-worker.js', { enabled: environment.production })
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
