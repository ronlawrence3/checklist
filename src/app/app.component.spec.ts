import {async, TestBed} from '@angular/core/testing';
import {RouterTestingModule} from '@angular/router/testing';
import {AppComponent} from './app.component';
import {MatToolbarModule} from '@angular/material/toolbar';
import {MatButtonModule} from '@angular/material/button';
import {MatSidenavModule} from '@angular/material/sidenav';
import {NoopAnimationsModule} from '@angular/platform-browser/animations';
import {MatIconModule} from '@angular/material/icon';
import {MatCardModule} from '@angular/material/card';
import {ChecklistModule} from './checklist/checklist.module';
import {Store} from '@ngrx/store';

describe('AppComponent', () => {
  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [
        RouterTestingModule,
        MatToolbarModule,
        MatButtonModule,
        MatSidenavModule,
        NoopAnimationsModule,
        MatToolbarModule,
        MatButtonModule,
        MatCardModule,
        MatSidenavModule,
        ChecklistModule,
        MatIconModule,
      ],
      declarations: [AppComponent],
      providers: [
        {provide: Store, useValue: {}}
      ]
    }).compileComponents();
  }));

  it('should create the app', () => {
    const fixture = TestBed.createComponent(AppComponent);
    const app = fixture.debugElement.componentInstance;
    expect(app).toBeTruthy();
  });

  it(`should have as title 'checklist'`, () => {
    const fixture = TestBed.createComponent(AppComponent);
    const app = fixture.debugElement.componentInstance;
    expect(app.title).toMatchSnapshot();
  });
});
