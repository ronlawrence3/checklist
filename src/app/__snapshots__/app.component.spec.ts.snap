// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should have as title 'checklist' 1`] = `"checklist"`;
